lua/nvim-tree/renderer/help.lua

```diff
@@ -1,27 +1,62 @@
+local log = require "nvim-tree.log"
+
 local M = {}
 
+local function shorten_lhs(lhs)
+  lhs = lhs:gsub("LeftMouse", "LM")
+  lhs = lhs:gsub("RightMouse", "RM")
+  lhs = lhs:gsub("MiddleMouse", "MM")
+  lhs = lhs:gsub("ScrollWheelDown", "SWD")
+  lhs = lhs:gsub("ScrollWheelUp", "SWU")
+  lhs = lhs:gsub("ScrollWheelLeft", "SWL")
+  lhs = lhs:gsub("ScrollWheelRight", "SWR")
+  return lhs
+end
+
 function M.compute_lines()
   local help_lines = { "HELP" }
   local help_hl = { { "NvimTreeRootFolder", 0, 0, #help_lines[1] } }
-  local mappings = vim.tbl_filter(function(v)
-    return (v.cb ~= nil and v.cb ~= "") or (v.action ~= nil and v.action ~= "")
-  end, require("nvim-tree.actions").mappings)
+
+  local buf_keymaps = vim.api.nvim_buf_get_keymap(vim.api.nvim_get_current_buf(), "n")
+
+  log.line("dev", "%s", vim.inspect(buf_keymaps))
+
   local processed = {}
-  for _, b in pairs(mappings) do
-    local cb = b.cb
-    local key = b.key
-    local name
-    if cb and cb:sub(1, 35) == require("nvim-tree.config").nvim_tree_callback("test"):sub(1, 35) then
-      name = cb:match "'[^']+'[^']*$"
-      name = name:match "'[^']+'"
-    elseif b.action then
-      name = b.action
+  for _, bkm in ipairs(buf_keymaps) do
+    local default_keymap = nil
+    for _, dkm in ipairs(require("nvim-tree.keymap").DEFAULT_KEYMAPS) do
+      if bkm.callback == dkm.callback then
```

Comment:
Very happy with this. It successfully maps user's on_attach mappings as well as defaults.

`inject_node` was not a problem as it returns an anonymous function that mapped well.